src/twisted/trial/_synctest.py

```diff
@@ -1015,10 +1015,17 @@ def getSkip(self):
         Return the skip reason set on this test, if any is set. Checks on the
         instance first, then the class, then the module, then packages. As
         soon as it finds something with a C{skip} attribute, returns that.
+        If the C{skip} attribute does not exist, look for C{__unittest_skip__}
+        and C{__unittest_skip_why__} attributes which are set by the standard
+        library L{unittest.skip} function.
         Returns L{None} if it cannot find anything. See L{TestCase} docstring
         for more details.
         """
-        return util.acquireAttribute(self._parents, 'skip', None)
+        skipReason = util.acquireAttribute(self._parents, 'skip', None)
+        if skipReason is None:
+            if getattr(self, "__unittest_skip__", None):
+                skipReason = getattr(self, "__unittest_skip_why__", None)
```

Comment:
Is this correct?

It looks like if it has `__unittest_skip__ == True and __unittest_skip_why__ == "" `  it will not skip

```suggestion
           # See if we have a stdlib skip condition.
            skipReason = getattr(self, "__unittest_skip__", None)
            if skipReason:
                # See if we have more details about the skip.
                skipReason = getattr(self, "__unittest_skip_why__", True) or True
```

Reply:
I implemented it like that to be consistent with the behavior in
* unittest: https://github.com/python/cpython/blob/515fce4fc4bb0d2db97b17df275cf90640017f56/Lib/unittest/case.py#L569
* pytest: https://github.com/pytest-dev/pytest/blob/952762207a2de76bf2eb254dee16f3b53d79a77b/src/_pytest/unittest.py#L92

Reply:
What I can read in cpython, is that if you have `__unittest_skip__` , `self._addSkip` is always called... with or without a reason.

But from what I can see in your implementation, `__unittest_skip_why__` will overwrite `__unittest_skip__`

So if you have `__unittest_skip__ == True ` and `__unittest_skip_why__ == False` the implementation will end up with   skipReason = False and the test is not skipped.

Reply:
You are misreading unittest.  `__unittest_skip__` is a boolean value, and `__unittest_skip_why__` is a string value.  Look at the pytest implementation as well.

Reply:
I changed the implementation here slightly to match what was done in pytest.

Reply:
My expectation is that if I have `@unittest.skipIf(True, '')` the test is skipped.
From what I can read in cpython code, this test is skipped.
From what I can read from the code in this branch, the test is not skipped.

Reply:
You are reading the unittest code wrong.  That's not how it behaves.
For this example:

```
from unittest import TestCase
from unittest import skipIf, main

class TestStringMethods(TestCase):

    @skipIf(True, '')
    def test_upper(self):
        print("RAN test_upper")
        self.assertEqual('foo'.upper(), 'FOO')

    @skipIf(False, '')
    def test_isupper(self):
        print("RAN test_isupper")
        self.assertTrue('FOO'.isupper())
        self.assertFalse('Foo'.isupper())

    @skipIf(True, '')
    def test_split(self):
        print("RAN test_split")
        s = 'hello world'
        self.assertEqual(s.split(), ['hello', 'world'])
        # check that s.split fails when the separator is not a string
        with self.assertRaises(TypeError):
            s.split(2)


if __name__ == '__main__':
    main()
```

Only the first argument to `skipIf` is used for skipping.  The second argument, even if it is an empty string, does not control whether skipping occurs or not.

Reply:
I updated this PR, and added more testcases with:

```
@skipIf(False, "")

@skipIf(True, "")
```

on both classes and functions.